vortex-flatbuffers/flatbuffers/vortex-serde/footer.fbs

```diff
@@ -1,24 +1,18 @@
-table FlatLayout {
-    begin: uint64;
-    end: uint64;
-}
-
-table NestedLayout {
+table Layout {
     children: [Layout];
     encoding: uint16;
-}
-
-union LayoutVariant {
-    FlatLayout,
-    NestedLayout,
-}
-
-table Layout {
-    layout: LayoutVariant;
+    metadata: [ubyte];
 }
 
 table Footer {
     layout: Layout;
+    row_count: uint64;
+}
+
+table Postscript {
+    schema_offset: uint64;
+    footer_offset: uint64;
 }
```

Comment:
IMO, should also define something like

```
struct EndOfFile {
    version: u16
    magic: [ubyte; 4]
}
```

this is the unevolvable part (last 8 bytes of the file must forever be these 8 bytes), but because the Postscript offset (from the end) will be derivable from `version`, we'll be actually able to add stuff to the Postscript (which would be guaranteed to be read in first read, whereas Footer eventually may not be)

Reply:
Sure I have done it but flatbuffer structs can't be top level elements. Fixed size arrays can't be members of tables so minimal thing you end up with is
```
struct Magic {
    value: [ubyte:4];
}

table EndOfFile {
    version: uint16;
    magic: Magic;
}
```
which is 20 bytes.
I'd say we serialize the last 8 bytes manually. have to make sure to pad the 6 bytes to 8